pkg/controller/util/endpoint/trigger_time_tracker.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2019 The Kubernetes Authors.
```

Comment:
nit:  put trigger_time_tracker file changes in a separate commit. (no need to include the controller changes)

Reply:
Will keep this in mind as the PR progresses, for now I decided to create another commit with the changes after this review so it was easier to see that diff, but let me know if it's better to keep commits structured around content.